packages/ra-ui-materialui/src/input/TextInput.stories.tsx

```diff
@@ -27,6 +27,38 @@ export const Basic = () => (
     </AdminContext>
 );
 
+export const Disabled = () => (
+    <AdminContext>
+        <Create
+            resource="posts"
+            record={{ id: 123, title: 'Lorem ipsum' }}
+            sx={{ width: 600 }}
+        >
+            <SimpleForm>
+                <TextInput source="title" disabled />
+                <TextInput source="name" disabled />
+                <FormInspector />
+            </SimpleForm>
+        </Create>
+    </AdminContext>
+);
+
+export const ReadOnly = () => (
+    <AdminContext>
+        <Create
+            resource="posts"
+            record={{ id: 123, title: 'Lorem ipsum' }}
+            sx={{ width: 600 }}
+        >
+            <SimpleForm>
+                <TextInput source="title" readOnly />
+                <TextInput source="name" readOnly />
```

Comment:
```suggestion
                
```

Reply:
It's voluntary. It's a request from JB: to be able to see input behavior with and without data.